qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobPushIT.java

```diff
@@ -0,0 +1,115 @@
+import io.zeebe.containers.cluster.ZeebeCluster;
+import io.zeebe.containers.engine.ContainerEngine;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.StreamSupport;
+import org.agrona.CloseHelper;
+import org.assertj.core.groups.Tuple;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.containers.Network;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+final class JobPushIT {
+  private final Network network = Network.newNetwork();
+  private final ZeebeCluster cluster =
+      ZeebeCluster.builder()
+          .withBrokersCount(1)
+          .withGatewaysCount(1)
+          .withPartitionsCount(1)
+          .withEmbeddedGateway(false)
+          .withImage(ZeebeTestContainerDefaults.defaultTestImage())
+          .build();
+
+  @SuppressWarnings("unused")
+  @RegisterExtension
+  final ContainerLogsDumper logsWatcher = new ContainerLogsDumper(cluster::getNodes);
+
+  @Container
+  private final ContainerEngine engine =
+      ContainerEngine.builder()
+          .withDebugReceiverPort(SocketUtil.getNextAddress().getPort())
+          .withCluster(cluster)
+          .withIdlePeriod(Duration.ofSeconds(5))
+          .build();
+
+  @AfterEach
+  void afterEach() {
+    CloseHelper.quietCloseAll(network);
+  }
+
+  @Test
+  void shouldStreamActivatedJobs() throws InterruptedException, TimeoutException {
+    // given
+    final var process =
+        Bpmn.createExecutableProcess("process")
+            .startEvent()
+            .serviceTask("task", t -> t.zeebeJobType("foo"))
+            .endEvent()
+            .done();
+
+    try (final var client = engine.createClient()) {
+      final var deploymentEvent =
+          client.newDeployResourceCommand().addProcessModel(process, "process.bpmn").send().join();
+
+      // when
+      final List<ActivatedJob> jobs = new ArrayList<>();
+      client
+          .newStreamJobsCommand()
+          .jobType("foo")
+          .consumer(jobs::add)
+          .timeout(Duration.ofMinutes(2))
+          .send();
+
+      client
+          .newCreateInstanceCommand()
+          .processDefinitionKey(deploymentEvent.getProcesses().get(0).getProcessDefinitionKey())
+          .requestTimeout(Duration.ofMinutes(1))
+          .send()
+          .join();
+
+      // then
+      // wait for the record to be available and consumed
+      engine.waitForIdleState(Duration.ofSeconds(30));
+      assertThat(records())
+          .as("job should become activated immediately")
+          .extracting(Record::getValueType, Record::getIntent)
+          .containsSubsequence(Tuple.tuple(ValueType.JOB_BATCH, JobBatchIntent.ACTIVATED));
+      assertThat(jobs).hasSize(1);
```

Comment:
:x: It's possible for the job batch activated record to be written, but the job to not have been sent or received by the stream consumer yet. Instead, I think it's perfectly fine to just use Awaitility and wait until we receive the expected job within 10 seconds or so. That means we probably don't need the whole ContainerEngine here.

A more complete alternative would be to wait for the job, complete it, and then assert that the process instance is completed. But I don't think it's necessary here.

Reply:
We'll wait #13684 to be merged to properly write assertions to the IT test.

Cc: @npepinpe